app/server/src/routes/course/course.controller.js

```diff
@@ -0,0 +1,68 @@
+const CourseModel = require("../../models/course/course.model");
+const EnrollmentModel = require("../../models/enrollment/enrollment.model");
+
+const CourseController = {
+  createCourse: async function (req, res) {
+    const {
+      course_name,
+      lecturer_id,
+      course_info,
+      course_chapters,
+      course_tags,
+    } = req.body;
+    const course = await CourseModel.createCourse(
+      course_name,
+      lecturer_id,
+      course_info,
+      course_chapters,
+      course_tags
+    );
+    res.status(201).send({ course });
+  },
+
+  getCourses: async function (req, res) {
+    const { keyword } = req.body;
+    const courses = await CourseModel.find({
+      course_name: { $regex: keyword, $options: "i" },
+    })
+      .populate("name rating image lecturer")
+      .populate({ path: "lecturer" });
+    return res.status(200).json({ courses });
+  },
+
+  createEnrollment: async function (req, res) {
```

Comment:
why is this in course controller?

Reply:
Because enrollment is closely related to courses, I had started its functions in the course controller. Created new controller for enrollment functions.